src/types.rs

```diff
@@ -1653,46 +2190,222 @@ impl TRS {
     /// Remove any determinism restriction `self` might be under.
     ///
     /// Return `true` if `self` was changed, otherwise `false`.
+    ///
+    /// See [`Deterministic System`] for more information.
```

Comment:
Go ahead and move this link up to `determinism` in the previous sentence so we can delete this sentence.

Also, nice job linking to more information here and elsewhere in this PR! :+1: